Snowflake.Data.Tests/UnitTests/Logger/EasyLoggerManagerTest.cs

```diff
@@ -0,0 +1,117 @@
+{
+    [TestFixture, NonParallelizable]
+    public class EasyLoggerManagerTest
+    {
+        
+        private const string InfoMessage = "Easy logging Info message";
+        private const string DebugMessage = "Easy logging Debug message";
+        private const string WarnMessage = "Easy logging Warn message";
+        private const string ErrorMessage = "Easy logging Error message";
+        private const string FatalMessage = "Easy logging Fatal message";
+        private static readonly string s_logsDirectory = Path.GetTempPath();
+
+        [ThreadStatic]
+        private static string t_directoryLogPath;
+        
+        [OneTimeTearDown]
+        public static void CleanUp()
+        {
+            RemoveEasyLoggingLogFiles();
+        }
+
+        [SetUp]
+        public void BeforeEach()
+        {
+            t_directoryLogPath = RandomLogsDirectoryPath();
+        }
+
+        [TearDown]
+        public void AfterEach()
+        {
+            EasyLoggerManager.Instance.ReconfigureEasyLogging(EasyLoggingLogLevel.Warn, t_directoryLogPath);
+        }
+        
+        [Test]
+        public void TestThatChangesLogLevel()
+        {
+            // arrange
+            var logger = SFLoggerFactory.GetLogger<SFBlockingChunkDownloaderV3>();
+            EasyLoggerManager.Instance.ReconfigureEasyLogging(EasyLoggingLogLevel.Warn, t_directoryLogPath);
+            
+            // assert
+            Assert.IsFalse(logger.IsDebugEnabled());
+            Assert.IsFalse(logger.IsInfoEnabled());
+            Assert.IsTrue(logger.IsWarnEnabled());
+            Assert.IsTrue(logger.IsErrorEnabled());
+            Assert.IsTrue(logger.IsFatalEnabled());
+
+            // act
+            EasyLoggerManager.Instance.ReconfigureEasyLogging(EasyLoggingLogLevel.Debug, t_directoryLogPath);
+
+            // assert
+            Assert.IsTrue(logger.IsDebugEnabled());
+            Assert.IsTrue(logger.IsInfoEnabled());
+            Assert.IsTrue(logger.IsWarnEnabled());
+            Assert.IsTrue(logger.IsErrorEnabled());
+            Assert.IsTrue(logger.IsFatalEnabled());
+        }
+
+        [Test]
+        public void TestThatLogsToProperFileWithProperLogLevelOnly()
+        {
+            // arrange
+            var logger = SFLoggerFactory.GetLogger<SFBlockingChunkDownloaderV3>();
+            EasyLoggerManager.Instance.ReconfigureEasyLogging(EasyLoggingLogLevel.Info, t_directoryLogPath);
+            
+            // act
+            logger.Info(InfoMessage);
+            logger.Debug(DebugMessage);
+            logger.Warn(WarnMessage);
+            logger.Error(ErrorMessage);
+            logger.Fatal(FatalMessage);
+
+            // assert
+            var logLines = File.ReadLines(FindLogFilePath(t_directoryLogPath));
+            Assert.That(logLines, Has.Exactly(1).Matches<string>(s => s.Contains(InfoMessage)));
+            Assert.That(logLines, Has.Exactly(1).Matches<string>(s => s.Contains(WarnMessage)));
```

Comment:
Could you sort asserts by log level?
Don't we want to check the results of different log levels?

Reply:
I sorted them. Our SFLogger allows to log only with levels: debug, info, warn, error and fatal. So it is currently impossible to create a log item with other log level.